src/artemis/__main__.py

```diff
@@ -51,17 +51,25 @@ class BlueskyRunner:
     current_status: StatusAndMessage = StatusAndMessage(Status.IDLE)
     last_run_aborted: bool = False
 
-    def __init__(self, RE: RunEngine) -> None:
+    def __init__(self, RE: RunEngine, skip_startup_connection=False) -> None:
         self.RE = RE
+        self.skip_startup_connection = skip_startup_connection
         if VERBOSE_EVENT_LOGGING:
             RE.subscribe(VerbosePlanExecutionLoggingCallback())
-        for plan in PLAN_REGISTRY:
-            PLAN_REGISTRY[plan]["setup"]()
+
+        if not self.skip_startup_connection:
+            for plan in PLAN_REGISTRY:
+                PLAN_REGISTRY[plan]["setup"]()
 
     def start(
         self, experiment: Callable, parameters: InternalParameters
     ) -> StatusAndMessage:
         artemis.log.LOGGER.info(f"Started with parameters: {parameters}")
+
+        if self.skip_startup_connection:
+            for plan in PLAN_REGISTRY:
```

Comment:
I would prefer if this only called setup() for the specified plan, rather than all of them

Reply:
Good spot

Reply:
yeah, if we start it running without the flag, we want it to be ready to run any plan we send to it. but if we don't want it to setup the devices on startup, it would be better if it only runs the setup for each plan as needed. that way, for example, if the simulator has some but not all the devices, we can still run plans that has devices the simulator does have, while not trying to connect to ones it doesn't have.

Reply:
Makes sense - new version does this now